### ISSUE TITLE
[Feature Request] TabContainer: make tab-select preventable

### ISSUE THREAD

**Issue by ilhan007** [SAP/ui5-webcomponents#5638]
### **Feature Request Description**
As I developer, I would like to be able to prevent the tab-select event.
Origin: https://github.com/SAP/ui5-webcomponents/issues/4883

### **Proposed Solution**
There are several examples in other components that we already implemented this behaviour.
It's based on. the `fireEvent` return value, which indicates if the event has been prevented in the event handler.

Example:

```js
/**
		 *
		 * Fired, when the notification icon is activated.
		 *
		 *
		 * @event sap.ui.webcomponents.fiori.ShellBar#notifications-click
		 * @allowPreventDefault
		 * @param {HTMLElement} targetRef dom ref of the activated element
		 * @public
		 */
		"notifications-click": {
			detail: {
				targetRef: { type: HTMLElement },
			},
		},
		
		
		const prevented = !this.fireEvent("notifications-click", {}, true);
```

### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

A clear and concise description of the impact/urgency of the required feature.

### **Stakeholder Info** (if applicable)

- Organization: {...} SAP, UI5 Web Components 4 React
- Business impact: {...}

**Comment by ilhan007** [comment]
Hello @SAP/ui5-webcomponents-topic-rd 
step by step we do allow preventing of the events with standard means - event.preventDefault to make web components behave as regular HTML elements. This request is regarding the TabContainer's tab-select event.

**Comment by gmkv** [comment]
Solved with #5661
